### PR TITLE
Fix docs generation command

### PR DESCRIPTION
**Pull request by thelovekesh**
## Summary

- Fix the issue due to the PHP patch version in version compares.
- Fix code issues that are causing PHP parser warnings.

## Checklist

- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by thelovekesh** [comment]
A few warnings are coming up while docs generation from the PHP parser.

```log
[03-Nov-2023 14:45:07 UTC] PHP Deprecated:  Invalid characters passed for attempted conversion, these have been ignored in /var/www/html/wp-content/plugins/amp/vendor/nikic/php-parser/lib/PhpParser/Node/Scalar/String_.php on line 93
Deprecated: Invalid characters passed for attempted conversion, these have been ignored in /var/www/html/wp-content/plugins/amp/vendor/nikic/php-parser/lib/PhpParser/Node/Scalar/String_.php on line 93
```

These are specifically due to:

`x0B` in

https://github.com/ampproject/amp-wp/blob/66b86680f6f1db819be2a0b37f5d5dee64cac201/includes/sanitizers/class-amp-base-sanitizer.php#L880

`xEF`, `xBB` and `xBF` in

https://github.com/ampproject/amp-wp/blob/66b86680f6f1db819be2a0b37f5d5dee64cac201/includes/validation/class-amp-validation-manager.php#L2344


This can also be due to an old PHP parser or can be some legitimate warnings.

File: docs/src/Cli/GenerateCommand.php [hunk 1]
```diff
@@ -48,7 +48,7 @@ final class GenerateCommand {
 	public function __invoke( $args, $assoc_args ) {
 		// Bail if current PHP version is not supported.
 		// Current combination of phpDocumentor and nikic/php-parser supports 7.4 or less.
-		if ( ! version_compare( phpversion(), '7.4', '<=' ) ) {
+		if ( version_compare( substr( phpversion(), 0, 3 ), '7.4', '>' ) ) {
```
Comment (westonruter) [inline docs/src/Cli/GenerateCommand.php]: Why the `substr()` here?

```suggestion
		if ( version_compare( phpversion(), '7.4', '>' ) ) {
```

Comment (thelovekesh) [reply docs/src/Cli/GenerateCommand.php]: version compare with is returning true in comparison due to patch version in it https://3v4l.org/H6Hnn#v7.4.33

Comment (westonruter) [reply docs/src/Cli/GenerateCommand.php]: Ah, then probably better to just use `8.0`:

```suggestion
		if ( version_compare( phpversion(), '8.0', '>=' ) ) {
```

Comment (thelovekesh) [reply docs/src/Cli/GenerateCommand.php]: Perfect ❤️

File: .github/dependabot.yml [hunk 2]
```diff
@@ -9,6 +9,12 @@ updates:
     time: "17:00"
     timezone: America/Los_Angeles
   open-pull-requests-limit: 10
+  groups:
+    wp-cli-packages:
+      patterns:
+        - "wp-cli/*"
+  ignore:
+    - dependency-name: phpdocumentor/reflection
```
Comment (westonruter) [inline .github/dependabot.yml]: Could you add a comment to this explaining why it is ignored?

Comment (thelovekesh) [reply .github/dependabot.yml]: Done in https://github.com/ampproject/amp-wp/pull/7670/commits/34bed2f08699c6fbf536243f61147ccf0770e525.

**Review by westonruter — APPROVED** [review]

**Pull request merged**